### ISSUE TITLE
Simple Task Timer

### ISSUE THREAD

**Issue by petersirka** [aziz/PlainTasks#247]
Hi contributors,
many thanks for this awesome plugin. I have a simple improvement for PlainTasks:

```html
[ ] some task without timer
[ ] some task with the timer `43` minutes (but I don't work on it)
[ ] some task with the timer `63` minutes @working (currently I'm working on it)
```

`@working` tag increases each minute `63` in its line. I don't know whether is it possible, but this feature will be really helpful. Thanks!

**Comment by vovkkk** [comment]
No, this is too simplistic to be realistic for plain text format. User may open several files each may have timers, files could be modified outside of ST — so it is not clear when exactly pause or stop, and resume, if it is not clear then most certainly user would get wrong/mistaken results.

We already have [pausable timetracking](https://github.com/aziz/PlainTasks/blob/master/messages/Tutorial.todo#L70-L76), first you `@start` with exact date and time, then you `@toggle` to pause (again date & time are mandatory), then `toggle` to resume, etc. Finally, when you done (or decide to cancel) you’ll see summary of time, e.g. `@lasted(1:03)` i.e. 63 minutes.

**Comment by petersirka** [comment]
Hi @vovkkk,
many thanks for quick anwser.

> User may open several files each may have timers, files could be modified outside of ST

It will be problem of the user. If the line won't contain`@working` then the timer won't be needed.

> if it is not clear then most certainly user would get wrong/mistaken results.

I think that no, because most of users work only with one editor (Sublime) and the timer increases only minutes in lines which they contain `@working` keyword.

Pausable timetracking is good, but not perfect.
Thanks :+1:

**Comment by vovkkk** [comment]
> It will be problem of the user. If the line won't contain `@working` then the timer won't be needed.

No, it is problem of code, because the question is how to control it? code would need look for appearance of the tag, start timer, then looking for disappearance of tag, stop/pause timer — all this in separate thread(s). My point was that more simplistic approach looks from user pov, more complecated it is from code pov, esp. when we talking about plain text format.



> most of users work only with one editor (Sublime) and the timer increases only minutes in lines which they contain `@working` keyword.

Git, Dropbox…

---

I mean, if you want to try implement this then go ahead, by all means, I only doubt that can be done in reliable manner.

**Comment by divinites** [comment]
Is it possible to have a simpler timer that only count days instead of minutes? for example, if @created(26-01-22 01:11) @due( +3), then after 1 day, It should show @due( +2). The timer is not necessarily running in the background, e.g. it might scan the text only once when a todo file is opened.

**Comment by petersirka** [comment]
I created for it a simple solution:
https://www.todomator.com

**Comment by divinites** [comment]
It is awesome! I think it is an independent software, not a plugin, right?

**Comment by petersirka** [comment]
Yes --> it's the independent software (web application). It's my alternative to PlainTaks but with similar features.
Thanks.

**Comment by aziz** [comment]
We can use `phantoms` that's introduced recently in sublime, to add auto updating info sections that are separate from the plain text file. for example an updating timer in front of the task. @vovkkk

**Comment by vovkkk** [comment]
I don’t see anything about phantoms in docs http://www.sublimetext.com/docs/3/api_reference.html
I see it mentioned in log http://www.sublimetext.com/3dev but no any reference, how to use it?

Anyway, I’m about to move to another apartment (next 2 weeks, a lot to set up and sort out) and want to implement auto-refresh in FileBrowser (do not want to hasten, so it may take event month maybe 2).

**Comment by divinites** [comment]
https://forum.sublimetext.com/t/dev-build-3118/21270

here is a sample for phantoms.

**Comment by vovkkk** [comment]
Cool.
I won’t promise anything, but yea we definitely could do sth with this.

**Comment by FichteFoll** [comment]
Can probably be closed now.

**Comment by vovkkk** [comment]
This is valid feature request; main reason it is still unsatisfied is https://github.com/SublimeTextIssues/Core/issues/1531

We should keep it open, hoping for reliable phantoms in future versions of ST.
